PSGSuite/Public/Classroom/Sync-GSCourse.ps1

```diff
@@ -0,0 +1,552 @@
+                        }    
+                    }
+                    New-GSCourse @Params -Name $Name -OwnerId $OwnerId -Id $CourseAlias -User $OwnerId -ErrorAction Stop -Verbose:$Verbose | ForEach-Object {
+                        $Course = New-Object PSObject -Property @{
+                            Name = $_.Name
+                            OwnerID = $Global:GSUserCache[$_.OwnerID].user
+                            ID = $_.ID
+                            Section = $_.Section
+                            Description = $_.Description
+                            DescriptionHeading = $_.DescriptionHeading
+                            Room = $_.Room
+                            CourseState = $_.CourseState
+                            Alias = @($CourseAlias)
+                            Student = @()
+                            Teacher = @()
+                        }
+                        $Course.Teacher += $Course.OwnerID
+                        $CourseCache[$Course.ID] = $Course
+                        $AliasCache[$CourseAlias] = $Course.ID
+                        $CourseID = $Course.ID
+                    }
+                } catch {
+                    if ($ErrorActionPreference -eq 'Stop') {
+                        $PSCmdlet.ThrowTerminatingError($_)
+                    } else {
+                        Write-Error $_
+                    }
+                }
+            } else {
+                write-warning "Unable to create course '$CourseAlias'. The -createCourse switch is not present."
+                # The course does not exist. No point executing the remaining logic.
+                Return
+            }      
+        }
+
+        # Check course users
+        $StudentsToAdd = @()
+        $StudentsToRemove = @()
+        $TeachersToAdd = @()
+        $TeachersToRemove = @()
+        $MembershipUpdated = $False
+
+        Compare-Object -ReferenceObject $Student -DifferenceObject $CourseCache[$CourseId].Student | ForEach-Object {
+            If ($_.SideIndicator -eq "<="){
+                $StudentsToAdd += $_.inputObject
+            } else {
+                $StudentsToRemove += $_.inputObject
+            }
+        }
+        $TempTeacher = $Teacher.Clone()
+        $TempTeacher = $TempTeacher += $OwnerID | Select-Object -Unique
+        Compare-Object -ReferenceObject $Teacher -DifferenceObject $CourseCache[$CourseId].Teacher | ForEach-Object {
+            If ($_.SideIndicator -eq "<="){
+                $TeachersToAdd += $_.inputObject
+            } else {
+                $TeachersToRemove += $_.inputObject
+            }
+        }
+
+        If ($StudentsToRemove.count){
+            If ($RemoveStudents){
+                Try {
+                    $MembershipUpdated = $True
+                    Remove-GSCourseParticipant -CourseId $CourseAlias -Student $StudentsToRemove -Confirm:$Confirm -ErrorAction Stop -Verbose:$Verbose | Out-Null
```

Comment:
I don't think there's an automatic variable of `$Confirm` and it's not being set anywhere.  I think the intent here, along with `SupportsShouldProcess` in the CmdletBinding is to pass through confirmation, but I don't think this will work.  Since `$Confirm` doesn't exist, I believe it will get cast to `$false` here so in effect you're always disabling confirmation.

I think if you just take `-Confirm:$Confirm` out on these 3 commands, it will act like you want.  Which, I believe is that if you execute Sync-GSCourse it will ask for confirmation in these 3 cases (the two instances of Remove, and the Export-CSV), and if you run Sync-GSCourse with `-Confirm:$false` then the 3 commands won't prompt for confirmation.

Reply:
Looks like the variable `$ConfirmPreference` gets set, according to that article I linked in the main comment.  You can also add `-WhatIf:$WhatIfPreference` to these commands _to be sure_ the WhatIf is passed through.